source/portable/NetworkInterface/STM32/NetworkInterface.c

```diff
@@ -0,0 +1,2373 @@
+            break;
+        }
+
+        if( prvGetPhyLinkStatus( pxInterface ) == pdFALSE )
+        {
+            FreeRTOS_debug_printf( ( "xNetworkInterfaceOutput: Link Down\n" ) );
+            break;
+        }
+
+        if( ( xMacInitStatus != eMacInitComplete ) || ( pxEthHandle->gState != HAL_ETH_STATE_STARTED ) )
+        {
+            FreeRTOS_debug_printf( ( "xNetworkInterfaceOutput: Interface Not Started\n" ) );
+            break;
+        }
+
+        // ETH_TxPacketConfigTypeDef
+        ETH_TxPacketConfig xTxConfig =
+        {
+            .CRCPadCtrl = ETH_CRC_PAD_INSERT,
+            .Attributes = ETH_TX_PACKETS_FEATURES_CRCPAD,
+        };
+
+        #if ipconfigIS_ENABLED( ipconfigDRIVER_INCLUDED_TX_IP_CHECKSUM )
+            xTxConfig.ChecksumCtrl = ETH_CHECKSUM_IPHDR_PAYLOAD_INSERT_PHDR_CALC;
+            xTxConfig.Attributes |= ETH_TX_PACKETS_FEATURES_CSUM;
+        #else
+            xTxConfig.ChecksumCtrl = ETH_CHECKSUM_DISABLE;
+        #endif
+
+        const EthernetHeader_t * const pxEthHeader = ( const EthernetHeader_t * const ) pxDescriptor->pucEthernetBuffer;
+
+        if( pxEthHeader->usFrameType == ipIPv4_FRAME_TYPE )
+        {
+            #if ipconfigIS_ENABLED( ipconfigUSE_IPv4 )
+                const IPPacket_t * const pxIPPacket = ( const IPPacket_t * const ) pxDescriptor->pucEthernetBuffer;
+
+                if( pxIPPacket->xIPHeader.ucProtocol == ipPROTOCOL_ICMP )
+                {
+                    #if ipconfigIS_ENABLED( ipconfigREPLY_TO_INCOMING_PINGS ) || ipconfigIS_ENABLED( ipconfigSUPPORT_OUTGOING_PINGS )
+                        ICMPPacket_t * const pxICMPPacket = ( ICMPPacket_t * const ) pxDescriptor->pucEthernetBuffer;
+                        #if ipconfigIS_ENABLED( ipconfigDRIVER_INCLUDED_TX_IP_CHECKSUM )
+                            pxICMPPacket->xICMPHeader.usChecksum = 0U;
+                        #endif
+                        ( void ) pxICMPPacket;
+                    #else
+                        FreeRTOS_debug_printf( ( "xNetworkInterfaceOutput: Unsupported ICMP\n" ) );
+                    #endif
+                }
+                else if( pxIPPacket->xIPHeader.ucProtocol == ipPROTOCOL_TCP )
+                {
+                    #if ipconfigIS_ENABLED( ipconfigUSE_TCP )
+                        TCPPacket_t * const pxTCPPacket = ( TCPPacket_t * const ) pxDescriptor->pucEthernetBuffer;
+
+                        /* #if defined( niEMAC_STM32HX ) && ipconfigIS_ENABLED( niEMAC_TCP_SEGMENTATION )
+                         *  xTxConfig.MaxSegmentSize = ipconfigTCP_MSS;
+                         *  xTxConfig.PayloadLen = pxDescriptor->xDataLength;
+                         *  xTxConfig.TCPHeaderLen = ( pxTCPPacket->xIPHeader.ucVersionHeaderLength & ( uint8_t ) 0x0FU );
+                         *  xTxConfig.Attributes |= ETH_TX_PACKETS_FEATURES_TSO;
+                         #endif */
+                        ( void ) pxTCPPacket;
```

Comment:
Untested feature of the Hx series. There should probably be a break in there if it is disabled, but I think disabled packet types are filtered out long before this stage anyways.

Reply:
Even though it's enabled, why do we need this part of code to get the `pxTCPPacket` pointer then do nothing with it?

Reply:
That code needs to be un-commented, then you will need the packet to adjust the tx settings before sending the packet. I just haven't been able to test it on an H5 because I don't have one